assembly/index.ts

```diff
@@ -0,0 +1,203 @@
+
+import { base64, runtime_api } from "near-runtime-ts";
+import { JSONEncoder as _JSONEncoder, JSON } from "assemblyscript-json";
+import { u128 } from "bignum";
+
+// Runtime functions
+// tslint:disable: no-unsafe-any
+
+@global
+class JSONEncoder extends _JSONEncoder {
+  encode<T>(name: string, val: T): void {
+    encode<T, JSONEncoder>(val, name, this);
+  }
+}
+
+type Usize = u64;
+//@ts-ignore
+@global
+function read_register(register_id: Usize, ptr: Usize): void {
+  runtime_api.read_register(register_id, ptr);
+}
+//@ts-ignore
+@global
+function register_len(register_id: Usize): Usize {
+  return runtime_api.register_len(register_id);
+}
+
+//@ts-ignore
+@global
+function input(register_id: Usize): void {
+  runtime_api.input(register_id);
+}
+//@ts-ignore
+@global
+function value_return(value_len: Usize, value_ptr: Usize): void {
+  runtime_api.value_return(value_len, value_ptr);
+}
+//@ts-ignore
+@global
+function panic(): void {
+  runtime_api.panic();
+}
+
+//@ts-ignore
+@global
+function getInput(): JSON.Obj { 
+  // Reading input bytes.
+  input(0);
+  let json_len = register_len(0);
+  if (json_len == U32.MAX_VALUE) {
+    panic();
+  }
+  let json = new Uint8Array(json_len as u32);
+  //@ts-ignore
+  read_register(0, <usize>json.buffer);
+  return <JSON.Obj> JSON.parse(json);
+}
+
+
+//@ts-ignore
```

Comment:
what does `@ts-ignore` do?  Why is it needed everywhere?

Reply:
It disables the editor from treating this as an error.  This particular one is because it doesn't like the decorator.  It's definitely better to provide the reason for why the line is ignored.  In this case most of them are caused because ts doesn't like that `T` can overlap with null and doesn't like to cast it when it can't determine the type.

Reply:
@willemneal given that almost everything here is annotated with `@ts-ignore` – is there a way to just disable this for whole file?

Also I wonder how much work it is to fork TS tooling for VS Code to work with all AssemblyScript peculiarities? We need to do it at some point I think.